cluster/kubernetes/resource/fluxhelmrelease.go

```diff
@@ -220,18 +327,109 @@ func interpretAsImage(m mapper) (image.Ref, ImageSetter, bool) {
+	cim := make(map[string]ContainerImageMap)
+	for k, v := range annotations {
+		switch {
+		case strings.HasPrefix(k, ImageRegistryPrefix):
+			container := strings.TrimPrefix(k, ImageRegistryPrefix)
+			i, _ := cim[container]
+			i.Registry = v
+			cim[container] = i
+		case strings.HasPrefix(k, ImageRepositoryPrefix):
+			container := strings.TrimPrefix(k, ImageRepositoryPrefix)
+			i, _ := cim[container]
+			i.Repository = v
+			cim[container] = i
+		case strings.HasPrefix(k, ImageTagPrefix):
+			container := strings.TrimPrefix(k, ImageTagPrefix)
+			i, _ := cim[container]
+			i.Tag = v
+			cim[container] = i
+		}
+	}
+	for k, _ := range cim {
+		i, _ := cim[k]
+		i.BasePath = ImageBasePath
+		cim[k] = i
+	}
+	return cim
+}
+
+func interpretMappedContainerImage(values map[string]interface{}, cim ContainerImageMap) (image.Ref, ImageSetter, bool) {
+	v, err := gabs.Consume(values)
+	if err != nil {
+		return image.Ref{}, nil, false
+	}
+
+	imageValue := v.Path(cim.Repository).Data()
+	if img, ok := imageValue.(string); ok {
+		if cim.Registry == "" && cim.Tag == "" {
+			if imgRef, err := image.ParseRef(img); err == nil {
+				return imgRef, func(ref image.Ref) {
+					v.SetP(ref.String(), cim.Repository)
+				}, true
+			}
+		}
+
+		switch {
+		case cim.Registry != "" && cim.Tag != "":
+			registryValue := v.Path(cim.Registry).Data()
+			if reg, ok := registryValue.(string); ok {
+				tagValue := v.Path(cim.Tag).Data()
+				if tag, ok := tagValue.(string); ok {
+					if imgRef, err := image.ParseRef(reg + "/" + img + ":" + tag); err == nil {
+						return imgRef, func(ref image.Ref) {
+							v.SetP(ref.Domain, cim.Registry)
+							v.SetP(ref.Image, cim.Repository)
+							v.SetP(ref.Tag, cim.Tag)
+						}, true
+					}
+				}
+			}
+		case cim.Registry != "":
+			registryValue := v.Path(cim.Registry).Data()
+			if reg, ok := registryValue.(string); ok {
+				if imgRef, err := image.ParseRef(reg + "/" + img); err == nil {
+					return imgRef, func(ref image.Ref) {
+						v.SetP(ref.Domain, cim.Registry)
+						v.SetP(ref.Name.Image+":"+ref.Tag, cim.Repository)
+					}, true
+				}
+			}
+		case cim.Tag != "":
+			tagValue := v.Path(cim.Tag).Data()
+			if tag, ok := tagValue.(string); ok {
+				if imgRef, err := image.ParseRef(img + ":" + tag); err == nil {
+					return imgRef, func(ref image.Ref) {
+						v.SetP(ref.Name.String(), cim.Repository)
+						v.SetP(ref.Tag, cim.Tag)
+					}, true
+				}
+			}
+		}
+	}
+
+	return image.Ref{}, nil, false
+}
+
 // Containers returns the containers that are defined in the
 // FluxHelmRelease.
 func (fhr FluxHelmRelease) Containers() []resource.Container {
 	var containers []resource.Container
-	// If there's an error in interpreting, return what we have.
-	_ = FindFluxHelmReleaseContainers(fhr.Spec.Values, func(container string, image image.Ref, _ ImageSetter) error {
+
+	containerSetter := func(container string, image image.Ref, _ ImageSetter) error {
```

Comment:
If you're giving this a name, `addContainer` might be better -- the implied analogy to `imageSetter` doesn't really work.